c_cxx/CMakeLists.txt

```diff
@@ -31,7 +31,16 @@ if(NOT ONNXRUNTIME_ROOTDIR)
 endif()
 
 #TODO: we should only need one of them.
-include_directories("${ONNXRUNTIME_ROOTDIR}/include" "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/session")
+include_directories(
+  "${ONNXRUNTIME_ROOTDIR}/include/"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/cpu"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/cuda"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/dnnl"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/nuphar"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/tensorrt"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/dml"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/providers/migraphx"
+  "${ONNXRUNTIME_ROOTDIR}/include/onnxruntime/core/session")
```

Comment:
So, it means we need to list all EPs here? Why?

If I  understood correctly, ONNXRUNTIME_ROOTDIR is where ORT is installed, not the source folder. At the installation step, should we already put all the EP specific header files into one place?